src/datamodel/validator/Uuid.cpp

```diff
@@ -10,3 +10,134 @@
+}
+}  // namespace
+
+namespace uprotocol::datamodel::validator::uuid {
+
+std::string_view message(Reason reason) {
+	switch (reason) {
+		case Reason::WRONG_VERSION:
+			return "Invalid UUID version";
+		case Reason::UNSUPPORTED_VARIANT:
+			return "Unsupported UUID variant";
+		case Reason::FROM_THE_FUTURE:
+			return "UUID timestamp is from the future";
+		case Reason::EXPIRED:
+			return "UUID has expired";
+		default:
+			return "Unknown reason";
+	}
+}
+
+ValidationResult isUuid(const uprotocol::v1::UUID uuid) {
+	uint64_t msb = uuid.msb();
+	uint64_t lsb = uuid.lsb();
+
+	uint8_t version = (msb >> 12) & 0xF;
+	if (version != 8) {
+		return {false, Reason::WRONG_VERSION};
+	}
+
+	uint8_t variant = (lsb >> 62) & 0x3;
+	if (variant != 2) {  // Variant should be 10 in binary
+		return {false, Reason::UNSUPPORTED_VARIANT};
+	}
+
+	std::chrono::system_clock::time_point timestamp = getUuidTimestamp(uuid);
+	std::chrono::system_clock::time_point current_time =
+	    std::chrono::system_clock::now();
+
+	if (timestamp > current_time) {
+		return {false, Reason::FROM_THE_FUTURE};
+	}
+
+	return {true, std::nullopt};
+}
+
+ValidationResult isExpired(const uprotocol::v1::UUID uuid,
+                           std::chrono::milliseconds ttl) {
+	auto [valid, reason] = isUuid(uuid);
+	if (!valid) {
+		return {false, reason};
+	}
+
+	uint64_t timestamp =
+	    (uuid.msb() >> 16) & 0xFFFFFFFFFFFF;  // Extract 48-bit timestamp
+	uint64_t current_time =
+	    std::chrono::duration_cast<std::chrono::milliseconds>(
+	        std::chrono::system_clock::now().time_since_epoch())
+	        .count();
+	if ((current_time - timestamp) > ttl.count()) {
+		return {true, Reason::EXPIRED};
+	}
+
+	return {false, std::nullopt};
+}
+
+uint8_t getVersion(const uprotocol::v1::UUID uuid) {
+	auto [valid, reason] = isUuid(uuid);
+	if (!valid) {
+		throw InvalidUuid(message(reason.value()));
+	}
+	return (uuid.msb() >> 12) & 0xF;
+}
+
+uint8_t getVariant(const uprotocol::v1::UUID uuid) {
+	auto [valid, reason] = isUuid(uuid);
+	if (!valid) {
+		throw InvalidUuid(message(reason.value()));
+	}
+	return (uuid.lsb() >> 62) & 0x3;
+}
+
+std::chrono::system_clock::time_point getTime(const uprotocol::v1::UUID uuid) {
+	auto [valid, reason] = isUuid(uuid);
+	if (!valid) {
+		throw InvalidUuid(message(reason.value()));
+	}
+	uint64_t timestamp = (uuid.msb() >> 16) & 0xFFFFFFFFFFFF;
+	return std::chrono::system_clock::time_point(
+	    std::chrono::milliseconds(timestamp));
+}
+
+std::chrono::milliseconds getElapsedTime(const uprotocol::v1::UUID uuid) {
+	auto [valid, reason] = isUuid(uuid);
+	if (!valid) {
+		throw InvalidUuid(message(reason.value()));
+	}
+
+	std::chrono::system_clock::time_point current_time =
+	    std::chrono::system_clock::now();
+	std::chrono::system_clock::time_point uuid_time = getTime(uuid);
+
+	return std::chrono::duration_cast<std::chrono::milliseconds>(current_time -
+	                                                             uuid_time);
```

Comment:
Minor: These are pretty long. You might consider adding
```
template <typename T>
using duration_cast = std::chrono::duration_cast<T>;
using milliseconds = std::chrono::milliseconds;
```
in the anonymous namespace.